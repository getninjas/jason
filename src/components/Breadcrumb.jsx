import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  steps: PropTypes.array.isRequired,
  active: PropTypes.number.isRequired,
};
export default class Breadcrumb extends Component {
  isActive(index) {
    return this.props.active === index;
  }

  handleActiveStyle(index) {
    return this.isActive(index) ? 'form__steps-item--active' : '';
  }

  render() {
    return (
      <ol className="form__steps space-box-medium">
        {
          this.props.steps.map((step, index) => {
            return (
              <li key={`breadcrumb-${index}`} className={`form__steps-item ${this.handleActiveStyle(index)}`}>
                {step.breadcrumbTitle}
              </li>
            );
          })
        }
      </ol>
    );
  }
}

Breadcrumb.propTypes = propTypes;
