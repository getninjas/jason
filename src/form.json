{
  "form": {
    "title": "Form Title",
    "databaseId": 81,
    "zipcodeUrlService": "http://www.mocky.io/v2/5afaebc92f00000e00f7c654",
    "steps": [
      {
        "title": "General data",
        "breadcrumbTitle": "Your request",
        "buttonText": "Start request",
        "headerMarkup": "<h2 class='widget__title section-title--high-contrast'>Lorem Ipsum is simply dummy text <small>It has survived not</small></h2>",
        "fields": [
          {
            "title": "Qual é o tipo de serviço que você precisa?",
            "placeholder": "Placeholder",
            "required": true,
            "id": "1_id",
            "name": "1_name_select",
            "type": "select",
            "value": 7117,
            "values": [
              {
                "databaseId": 7117,
                "value": "Construção"
              },
              {
                "databaseId": 7118,
                "value": "Reformas"
              },
              {
                "databaseId": 7119,
                "value": "Instalações"
              },
              {
                "databaseId": 7120,
                "value": "Troca"
              },
              {
                "databaseId": 7121,
                "value": "Outros"
              }
            ]
          },
          {
            "title": "Especifique melhor a metragem do serviço",
            "required": true,
            "id": "2_id",
            "name": "2_name_text",
            "type": "text",
            "placeholder": "Bora pra action",
            "value": "AHUAHuhhHUHAU",
            "values": []
          },
          {
            "title": "Informações adicionais",
            "required": false,
            "type": "textarea",
            "id": "3_id",
            "name": "3_name_textarea",
            "placeholder": "Descreva o que você precisa",
            "value": null,
            "values": []
          },
          {
            "required": true,
            "title": "Você precisa de serviços adicionais?",
            "placeholder": "Você precisa de serviços adicionais?",
            "id": "0_id",
            "name": "1_name_select",
            "type": "checkbox",
            "value": null,
            "values": [
              {
                "databaseId": 6792,
                "value": "Cozinhar"
              },
              {
                "databaseId": 7639,
                "value": "OTHER"
              }
            ]
          }
        ]
      },
      {
        "title": "User Data",
        "breadcrumbTitle": "Your data",
        "buttonText": "Finish",
        "headerMarkup": "<div class='form__alert-message space-element-large'><img src='https://www.getninjas.com.br/assets/request/phone_call.svg' class='form__icon' alt='Lorem Ipsum is simply dummy text of the printing and typesetting industry.'><h2 class='form__info'>Almost there<small>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's<strong>standard dummy text ever since the 1500s</strong></small></h2></div>",
        "fields": [
          {
            "title": "Zipcode",
            "required": true,
            "type": "zipcode",
            "id": "zip_code",
            "name": "zipcode",
            "placeholder": "00000-000",
            "value": "04904-060",
            "wrapperClassName": "form__field input",
            "values": []
          },
          {
            "title": "DDD + Celular",
            "id": "phone",
            "name": "phone",
            "type": "phone",
            "placeholder": "(__) _____-____",
            "required": true,
            "value": "(11) 96311-2689",
            "values": []
          }
        ]
      }
    ]
  }
}
