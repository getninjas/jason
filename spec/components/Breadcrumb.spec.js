import React from 'react';
import renderer from 'react-test-renderer';
import Breadcrumb from '../../src/components/Breadcrumb';
import { form } from '../../src/form.json';
import { enzymeConfig, shallow } from '../enzymeConfig';

enzymeConfig();

describe('Breadcrumb', () => {
  it('renders defaultProps', () => {
    const component = renderer.create(<Breadcrumb active={0} steps={form.steps} />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('.isActive', () => {
    describe('when first step is active', () => {
      const component = shallow(<Breadcrumb active={0} steps={form.steps} />);

      it('returns true for first step', () => {
        const result = component.instance().isActive(0);

        expect(result).toBe(true);
      });

      it('returns false for second step', () => {
        const secondResult = component.instance().isActive(1);

        expect(secondResult).toBe(false);
      });
    });

    describe('when second step is active', () => {
      const component = shallow(<Breadcrumb active={1} steps={form.steps} />);

      it('returns false for first step', () => {
        const result = component.instance().isActive(0);

        expect(result).toBe(false);
      });

      it('returns true for second step', () => {
        const secondResult = component.instance().isActive(1);

        expect(secondResult).toBe(true);
      });
    });
  });

  describe('.handleActiveStyle', () => {
    describe('when first step is active', () => {
      const component = shallow(<Breadcrumb active={0} steps={form.steps} />);

      it('returns active class .form__steps-item--active', () => {
        const result = component.instance().handleActiveStyle(0);

        expect(result).toEqual('form__steps-item--active');
      });

      it('does not return any class', () => {
        const result = component.instance().handleActiveStyle(1);

        expect(result).toEqual('');
      });
    });
  });

  describe('when second step is active', () => {
    const component = shallow(<Breadcrumb active={1} steps={form.steps} />);

    it('expects the first breadcrumb to be inactive', () => {
      expect(component.find('li').at(0).hasClass('form__steps-item--active')).toBe(false);
    });

    it('expects the second breadcrumb to be active', () => {
      expect(component.find('li').at(1).hasClass('form__steps-item--active')).toBe(true);
    });
  });

  describe('when third step is active', () => {
    const component = shallow(<Breadcrumb active={2} steps={form.steps} />);

    it('expects the first breadcrumb to be inactive', () => {
      expect(component.find('li').at(0).hasClass('form__steps-item--active')).toBe(false);
    });

    it('expects the second breadcrumb to be inactive', () => {
      expect(component.find('li').at(1).hasClass('form__steps-item--active')).toBe(false);
    });

    it('expects the second breadcrumb to be active', () => {
      expect(component.find('li').at(2).hasClass('form__steps-item--active')).toBe(true);
    });
  });
});
