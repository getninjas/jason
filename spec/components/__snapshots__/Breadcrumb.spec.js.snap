// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb renders defaultProps 1`] = `
<ol
  className="form__steps space-box-medium"
>
  <li
    className="form__steps-item form__steps-item--active"
  >
    Seu pedido
  </li>
  <li
    className="form__steps-item "
  >
    Seus dados
  </li>
</ol>
`;
